### PR TITLE
[Frontend] Remove -enable-operator-designated-types

### PR DESCRIPTION
**Pull request by harlanhaskins**
This patch adds -enable-operator-designated-types to the set of flags
serialized at the top of a .swiftinterface file, as it's used in the
standard library.

We need a better, long-term solution to staging flags like this one, but
this helps get the standard library building lazily from an interface.

### CODE REVIEW

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test macOS platform

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Review by jrose-apple — APPROVED** [review]

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Review by graydon — APPROVED** [review]
LGTM

**Comment by harlanhaskins** [comment]
@rudkx I think that's the better way to go.

**Comment by harlanhaskins** [comment]
@swift-ci please test

**Review by graydon — APPROVED** [review]
Looks just as good as before!

File: include/swift/Option/FrontendOptions.td [hunk 1]
```diff
@@ -393,10 +393,6 @@ def solver_disable_shrink :
 def disable_constraint_solver_performance_hacks : Flag<["-"], "disable-constraint-solver-performance-hacks">,
   HelpText<"Disable all the hacks in the constraint solver">;
 
-def enable_operator_designated_types :
-  Flag<["-"], "enable-operator-designated-types">,
-  HelpText<"Enable operator designated types">;
-
```
Comment (rudkx) [inline include/swift/Option/FrontendOptions.td]: This isn't quite what I had in mind. You could leave this in and revert the change to the test that explicitly uses this, and then have ParseStdlib also set the flag below.

**Review by rudkx — APPROVED** [review]

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test and merge

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test
